service/controller/cloudconfig/cloud_config.go

```diff
@@ -85,10 +87,17 @@ type CloudConfig struct {
 	dockerhubToken  string
 	ignitionPath    string
 	k8sAPIExtraArgs []string
+	proxy           proxyConfig
 	registryMirrors []string
 	ssoPublicKey    string
 }
 
+type proxyConfig struct {
```

Comment:
Maybe we could drop this and just use `ProxyConfig` (the exported one) in the `CloudConfig` struct?

Reply:
The exported one has `[]string` type for `NoProxy` where the private one already has the values joined so it's just a string.

Reply:
They could also be joined here https://github.com/giantswarm/kvm-operator/blob/02dc26a6dafc069275ac11a1f8303d9f800675af/service/controller/cloudconfig/master_template.go#L52 or in the template itself, but it's fine if you prefer this way.

Reply:
I will leave it as it is for now cause I don't have time to retest now and we need it asap. I can clean it up in a future commit if that's ok with you :)